packages/next/build/webpack-config.ts

```diff
@@ -436,6 +437,17 @@ export default async function getBaseWebpackConfig(
     loggedSwcDisabled = true
   }
 
+  if (
+    !loggedIgnoredCompilerOptions &&
+    !useSWCLoader &&
+    config.compiler != null
```

Comment:
Usually `!=` is a typo and it really should check for `!==`. But I think you intended to use it for undefined and null. Can we change it to:

```suggestion
    config.compiler
```